pkg/components/gtime/gtime_test.go

```diff
@@ -8,24 +8,26 @@ import (
 )
 
 func TestParseInterval(t *testing.T) {
+	now := time.Now()
+
 	tcs := []struct {
 		interval string
 		duration time.Duration
 		err      error
 	}{
-		{interval: "1d", duration: time.Hour * 24},
-		{interval: "1w", duration: time.Hour * 24 * 7},
-		{interval: "2w", duration: time.Hour * 24 * 7 * 2},
-		{interval: "1y", duration: time.Hour * 24 * 7 * 365},
-		{interval: "5y", duration: time.Hour * 24 * 7 * 365 * 5},
-		{interval: "1M", err: errors.New("time: unknown unit M in duration 1M")},
+		{interval: "1d", duration: now.Sub(now.AddDate(0, 0, -1))},
+		{interval: "1w", duration: now.Sub(now.AddDate(0, 0, -7))},
+		{interval: "2w", duration: now.Sub(now.AddDate(0, 0, -14))},
+		{interval: "1M", duration: now.Sub(now.AddDate(0, -1, 0))},
+		{interval: "1y", duration: now.Sub(now.AddDate(-1, 0, 0))},
+		{interval: "5y", duration: now.Sub(now.AddDate(-5, 0, 0))},
 		{interval: "invalid-duration", err: errors.New("time: invalid duration invalid-duration")},
 	}
 
 	for i, tc := range tcs {
 		t.Run(fmt.Sprintf("testcase %d", i), func(t *testing.T) {
 			res, err := ParseInterval(tc.interval)
-			if err != nil && err.Error() != tc.err.Error() {
+			if err != nil && (tc.err == nil || err.Error() != tc.err.Error()) {
```

Comment:
It's better if we rewrite this according to the PR I made to your repo, so we use the standard testify assertions. It also has much cleaner logic.